### PR TITLE
Feat/vue init

### PR DESCRIPTION
**Pull request by afsalz**
## Type of change

- [ ]  Patch: Bug (non-breaking change which fixes an issue)
- [x]  Minor: New feature (non-breaking change which adds functionality)
- [ ]  Major: Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Checklist

- [ ] This is something we need to do.

### CODE REVIEW

File: packages/vue/src/dotlottie.ts [hunk 1]
```diff
@@ -0,0 +1,161 @@
...
+    backgroundColor: { type: String, required: false },
+    data: { type: String, required: false },
+    loop: { type: Boolean, required: false },
+    mode: { type: String as () => Mode, required: false },
+    renderConfig: { type: Object, required: false },
+    segments: { type: Array as unknown as () => [number, number], required: false },
+    speed: { type: Number, required: false },
+    src: { type: String, required: false },
+    useFrameInterpolation: { type: Boolean, required: false },
+  },
```
Comment (theashraf) [inline packages/vue/src/dotlottie.ts:35]: can we infer the `type` of each prop from the dotlottie-web `Config` type. i wonder if we can just pass the type of the `DotLottieVueProps` created earlier 👀

File: packages/vue/README.md [hunk 2]
```diff
@@ -0,0 +1,125 @@
...
+</script>
+
+<template>
+  <DotLottieVue style="height: 500px; width: 500px" autoplay loop src="https://path-to-lottie.lottie" />
+</template>
+```
+
+## APIs
+
+### DotLottieVue Props
```
Comment (theashraf) [inline packages/vue/README.md:58]: missing `useFrameInterpolation` prop

File: apps/dotlottie-vue-example/src/App.vue [hunk 3]
```diff
@@ -0,0 +1,67 @@
+<script setup lang="ts">
+import { Ref, onMounted, ref } from 'vue';
+import { DotLottieVue } from '@lottiefiles/dotlottie-vue';
+const player: Ref<typeof DotLottieVue | null> = ref(null);
+const loop = ref(true);
+
+function toggleLoop() {
+  loop.value = !loop.value;
```
Comment (theashraf) [inline apps/dotlottie-vue-example/src/App.vue:8]:
```suggestion
  const dotLottie = playerRef.getDotLottieInstance();
  dotLottie.setLoop(!dotLottie.loop);
```

File: packages/vue/README.md [hunk 4]
```diff
@@ -0,0 +1,125 @@
+# @lottiefiles/dotlottie-react
```
Comment (theashraf) [inline packages/vue/README.md:1]:
```suggestion
# @lottiefiles/dotlottie-vue
```

**Review by theashraf — COMMENTED** [review]
Let's include dotlottie-vue in the `.size-limit.cjs`  config
```
{
    name: '@lottiefiles/dotlottie-vue',
    path: 'packages/vue/dist/index.js',
    import: '*',
    modifyWebpackConfig,
  },
```

File: packages/vue/src/dotlottie.ts (hunk 1 above)
Comment (afsalz) [reply packages/vue/src/dotlottie.ts:35]: these are not typescript types. its vue types feature, the first param of setup function takes the  `setup(props: DotLottieVueProps,..)` typescript types. Looks like both needs to be there to work. Also can't compile if both doesn't match haha

File: packages/vue/README.md [hunk 5]
```diff
@@ -0,0 +1,126 @@
...
+      console.log('Frame:', currentFrame)
+    });
+  }
+})
+</script>
+
+<template>
+  <DotLottieVue autoplay loop ref="player" src="path-to-lottie.lottie" />
+</template>
+```
```
Comment (theashraf) [inline packages/vue/README.md:96]: we need to update this example
```suggestion
```vue
<script setup>
import { onMounted, ref, watch } from 'vue';
import { DotLottieVue } from '@lottiefiles/dotlottie-vue'
const playerRef = ref(null);
const dotLottie = playerRef?.value.getDotLottieInstance();

onMounted(() => {
    dotLottie.addEventListener('pause', () => {
      console.log('paused')
    });
    dotLottie.addEventListener('frame', ({ currentFrame }) => {
      console.log('Frame:', currentFrame)
    });
  }
})
</script>

<template>
  <DotLottieVue autoplay loop ref="player" src="path-to-lottie.lottie" />
</template>
```
```

File: packages/vue/README.md [hunk 6]
```diff
@@ -76,22 +76,23 @@ import { DotLottieVue } from '@lottiefiles/dotlottie-vue'
 <script setup>
 import { onMounted, ref, watch } from 'vue';
 import { DotLottieVue } from '@lottiefiles/dotlottie-vue'
-const player = ref(null);
+const playerRef = ref(null);
 
 onMounted(() => {
   if (player.value) {
-    player.value.addEventListener('pause', () => {
+    const dotLottie = player.value.getDotLottieInstance();
```
Comment (theashraf) [inline packages/vue/README.md:83]:
```suggestion
    const dotLottie = playerRef.value.getDotLottieInstance();
```

Comment (theashraf) [reply packages/vue/README.md:83]: a small typo

File: packages/vue/README.md [hunk 7]
```diff
@@ -76,22 +76,23 @@ import { DotLottieVue } from '@lottiefiles/dotlottie-vue'
 <script setup>
 import { onMounted, ref, watch } from 'vue';
 import { DotLottieVue } from '@lottiefiles/dotlottie-vue'
-const player = ref(null);
+const playerRef = ref(null);
 
 onMounted(() => {
   if (player.value) {
```
Comment (theashraf) [inline packages/vue/README.md:82]:
```suggestion
  if (playerRef.value) {
```

**Review by theashraf — APPROVED** [review]
LGTM 🚀

**Pull request merged**